### PR TITLE
Add recipe for geiser-mit

### PR DESCRIPTION
**Pull request by jaor**
-------

### Brief summary of what the package does

GNU/MIT Scheme support for Geiser.  This is simply extracted from the
geiser package.

We're extracting each of the Scheme implementations supported by
Geiser to their own separate repo in https://gitlab.com/emacs-geiser/,
so there is little new code here.  Once this package enters MELPA,
we'll update the geiser package and delete these files.

We're going to reapeat this trick with 7 or 8 more supported schemes:
this one is for testing the waters. Is it okay to submit all the other
new packages as a single PR once they're ready, or do you prefer that
we go one by one?

Thanks!

### Direct link to the package repository

https://gitlab.com/emacs-geiser/mit

### Your association with the package

Maintainer.

### Relevant communications with the upstream package maintainer

None needed

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

### CODE REVIEW

File: recipes/geiser-mit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+(geiser-mit
+ :fetcher gitlab
+ :repo "emacs-geiser/mit"
+ :files ("geiser-mit.el" ("src" "src/*")))
```
Comment (riscy) [inline recipes/geiser-mit:4]: Possible opportunity to make this recipe flexible to growth:
```suggestion
 :files (:defaults ("src" "src/*")))
```

**Review by riscy — COMMENTED** [review]
> Is it okay to submit all the other new packages as a single PR once they're ready, or do you prefer that we go one by one?

I have a small preference for separate PRs, but if they're all closely related -- which I'd expect -- a single PR is fine.

I think since we're dealing with legacy code it may be fine to grandfather this particular issue in, but I notice two functions with names outside the package namespace are autoloaded: `run-mit` and `switch-to-mit`.  package-lint doesn't know to complain about these, but would if it could.  If it won't be a highly disruptive breaking change, it might be worth renaming these to fit the namespace.  Otherwise, for example, there's nothing stopping someone from blithely creating a package called `run-mit`.

**Comment by jaor** [comment]
On Sun, Mar 14 2021, Chris Rayner wrote:

> I think since we're dealing with legacy code it may be fine to grandfather this particular issue in, but I
> notice two functions with names outside the package namespace are autoloaded: run-mit and switch-to-mit.
> package-lint doesn't know to complain about these, but would if it could. If it won't be a highly disruptive
> breaking change, it might be worth renaming these to fit the namespace. Otherwise, for example, there's
> nothing stopping someone from blithely creating a package called run-mit.

that would be a disruptive change for users of the current geiser
package, which is in MELPA and exports them.  i'd rather not cause more
trouble than needed to existing users, who would already have to adapt
to the new layout.  do you feel very strongly against exporting them?
(taking into account that they are already claimed by geiser)

**Comment by jaor** [comment]
On Sun, Mar 14 2021, Chris Rayner wrote:

> @riscy commented on this pull request.
>
>     Is it okay to submit all the other new packages as a single PR once they're ready, or do you prefer that
>     we go one by one?
>    
> I have a small preference for separate PRs, but if they're all closely related -- which I'd expect -- a
> single PR is fine.

every extracted package means a new core geiser release and a new
warning that things are changing.  for users of more than one
implementation (which in scheme-land is not uncommon), a single PR would
be easier... so if your preference is really small, i'd go for one :)

**Comment by riscy** [comment]
> do you feel very strongly against exporting them?  (taking into account that they are already claimed by geiser)

I do but I think there's some room for compromise -- for example, would you be willing to rewrite these as deprecated functions that let users know about the deprecation?  This moment may be a great chance to do so, because your users will already have to be doing a little bit of migration work by the sounds of it.

Also have you considered using `:defaults` in your recipe?  This would give you some room to expand to e.g. a multi-file package in the recipe without another PR.  Not a blocker obviously.

**Comment by riscy** [comment]
If you let me know about the `:defaults` in the recipe this can still be a quick merge.  The namespacing can be grandfathered, even though you had to hear me complain about it. :)

**Comment by jaor** [comment]
On Sun, Mar 21 2021, Chris Rayner wrote:

>     do you feel very strongly against exporting them? (taking into account that they are already claimed by
>     geiser)
>    
> I do but I think there's some room for compromise -- for example,
> would you be willing to rewrite these as deprecated functions that let
> users know about the deprecation? This moment may be a great chance to
> do so, because your users will already have to be doing a little bit
> of migration work by the sounds of it.

yes, that sounds sensible.  these run-* and switch-to-* functions are
defined by a shared macro in the base geiser package, so i can easily
deprecate all at once.  i'll do that.

> Also have you considered using :defaults in your recipe? This would
> give you some room to expand to e.g. a multi-file package in the
> recipe without another PR. Not a blocker obviously.

good idea.  i've pushed a new version using :defaults.  please, let me
know if i messed anything up.

**Comment by riscy** [comment]
The build looks good!  I'll go ahead and merge.

**Comment by jaor** [comment]
thanks!

**Pull request merged**